### PR TITLE
Issue #3091 Fail when json configuration is invalid

### PR DESCRIPTION
**Pull request by gbraad**
edit `allinstances.json` and put in an erroneous character.

### CODE REVIEW

**Comment by gbraad** [comment]
Codecov is so handy, that all PRs are now turned red.
And what benefit do we actually get? If no actions are taken, these 'integrations' are useless.

**Comment by praveenkumar** [comment]
@gbraad allinstance config is something we create and not expect user to change anything by hand :) I think that's why till now we didn't have this kind of bug.

File: pkg/minishift/config/allinstancesconfig.go [hunk 1]
```diff
@@ -83,6 +84,10 @@ func (cfg *GlobalConfigType) read() error {
 		return err
 	}
 
+	if !json.Valid(raw) {
+		return errors.New("Invalid JSON")
+	}
+
```
Comment (praveenkumar) [inline pkg/minishift/config/allinstancesconfig.go]: :+1:

**Comment by agajdosi** [comment]
@gbraad Issue with codecov was fixed, so it is not true that all PRs are red because of Codecov. But I understand you, I took a look and it seems we could define threshold for the commit status (https://docs.codecov.io/docs/commit-status), so if the coverage drops a little it won't fail the PR status.

**Comment by agajdosi** [comment]
Created #3093.

**Comment by agajdosi** [comment]
Oh, and on #2819 it looked like nobody was against it. If even -2% or 5% drops are ok, we can turn the reports completely off and keep just the comments about coverage. Or we can even remove it completely, lets discuss the actions we take on https://github.com/minishift/minishift/issues/3093.

**Pull request merged**